app/lib/experiments-client.coffee

```diff
@@ -0,0 +1,206 @@
+      intervention: "i"
+      classification: "c"
+      "-question-": "qu"
+      "-statement-": "st"
+      gamisation: "gam"
+      valued: "val"
+      learning: "lrn"
+    }
+    simple_data = {}
+    for field in fields_to_include
+      if field of data
+        fieldname_to_use = field
+        if field of fieldname_shortening_map
+          fieldname_to_use = fieldname_shortening_map[field]
+        value_to_use = JSON.parse(JSON.stringify(data[field])) # deep clone
+        if typeof value_to_use is 'string'
+          value_to_use = @replaceAllSubStringsByRegEx value_to_use, value_compression_map
+        else if typeof value_to_use is 'object'
+          for subvalue, index in value_to_use
+            if typeof subvalue is 'string'
+              value_to_use[index] = @replaceAllSubStringsByRegEx subvalue, value_compression_map
+        simple_data[fieldname_to_use] = value_to_use
+    simple_data
+
+  replaceAllSubStringsByRegEx: (string_to_modify, value_compression_map) ->
+    for regex_str, replacement of value_compression_map
+      regex = new RegExp regex_str,"gi"
+      string_to_modify = string_to_modify.replace regex, replacement
+    string_to_modify
+
+  # .. and if it's too long, we give up.
+  safeJSON: (simplified_data) ->
+    if simplified_data
+      jsonString = JSON.stringify(simplified_data)
+      if jsonString.length >= 512
+        jsonString = JSON.stringify {
+          error: "too_much_data"
+          details: "The data passed to Geordi was too large to store, at #{jsonString.length} characters (maximum is 512)."
+        }
+      jsonString
+    else
+      null
+
+  # experimentState == "experimentStart", "classificationStart", "classificationEnd", "interventionStart", "interventionEnd", or "experimentEnd"
+  logExperimentState: (geordi, experimentState, data) ->
+    simple_data = null
+    if data?
+      simple_data = @simplifyParticipantDataForLogging data
+    geordi?.logEvent
+      type: "experiment"
+      relatedID: experimentState
+      data: @safeJSON simple_data
+
+  # check if this project has an experiment enabled. Return experiment_name if so.
+  #
+  # (If it does, it should be notifying experiment server of its classifications, 
+  #  which will register it or progress it)
+  checkForExperiment: (project_slug) ->
+    enabled_experiments = @getEnabledExperiments()
+    if project_slug of enabled_experiments
+      enabled_experiments[project_slug]
+
+  # type = "classification" or "intervention", and id is the classification_id or intervention_id correspondingly
+  postDataToExperimentServer: (interventionMonitor, geordi, experiment_name, user_id, session_id, type, id) ->
+    @ensureEnvironmentSet()
+
+    if not session_id?
+      session_id = getSessionID()
+
+    post_path = "/experiment/#{experiment_name}/user/#{user_id}/session/#{session_id}/#{type}/#{id}"
+    post_url = "#{@EXPERIMENT_SERVER_URL_TO_USE}#{post_path}"
+
+    request = new XMLHttpRequest()
```

Comment:
We have the polyfill setup for fetch. It's the newer spec and it uses Promises: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API